mbedtls/src/x509/mbedtls_wrappers.rs

```diff
@@ -0,0 +1,241 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * Licensed under the GNU General Public License, version 2 <LICENSE-GPL or
+ * https://www.gnu.org/licenses/gpl-2.0.html> or the Apache License, Version
+ * 2.0 <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0>, at your
+ * option. This file may not be copied, modified, or distributed except
+ * according to those terms. */
+
+#[cfg(not(feature = "std"))]
+use crate::alloc_prelude::*;
+use mbedtls_sys::*;
+
+use crate::private::UnsafeFrom;
+use crate::{forward_mbedtls_calloc, forward_mbedtls_free};
+//use core::ops::{Deref, DerefMut};
+use mbedtls_sys::types::raw_types::c_void;
+use core::ptr::drop_in_place;
+use core::marker::PhantomData;
+use crate::x509::Certificate;
+use crate::error::{Error, Result};
+use core::ptr::NonNull;
+
+#[repr(transparent)]
+pub struct MbedtlsBox<T> {
+    inner: NonNull<T>
+}
```

Comment:
You can put the generic stuff (`MbedtlsBox<T>`, `MbedtlsList<T>`, `Drop`) in here (it should be up one level in the module hierarchy), but the rest should go in `certificate.rs`.